langchain4j-vertex-ai-gemini/src/test/java/dev/langchain4j/model/vertexai/VertexAiGeminiChatModelIT.java

```diff
@@ -111,4 +121,148 @@ void should_allow_custom_generativeModel_and_generationConfig() throws IOExcepti
         // then
         assertThat(response.content().text()).contains("Berlin");
     }
+
+    @Test
+    void should_accept_text_and_image_from_public_url() {
+
+        // given
+        Image image = Image.builder()
+                .url(CAT_IMAGE_URL)
+                .mimeType("image/png")
```

Comment:
Hi @glaforge, would be awesome if VertexAI SDK could detect mime type automatically (like OpenAI does) without forcing end user to specify it manually

Reply:
Yeah this is something we could do, but would be nice to keep it as simple as possible.

I've just experimented a bit and it seems that second part of mime type does not matter, it should start with `image/` hovewer. Even if the image is png, I can send `image/jpeg` and it works fine. I am not sure if this is a feature or a bug that we can abuse, but we could also just set the default mime type to `image/png` or whatever and it should work.

Another option would be to open a connection and get mime type using `url.openConnection().getContentType()`, but that might not always work...

Reply:
I have added simple mime type detection logic (by file extension, for popular image formats).
Please note that `PartMaker.fromMimeTypeAndData()` javadoc says that only `image/png` and `image/jpeg` are accepted, but it seems to accept other image types as well. So I have added mapping for many formats, not just `png` and `jpeg`, to make it easier for the user.